File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int[] freq = new int[256];
        Arrays.fill(freq, 0);
        int n = s.length();
        int i = 0;
        int j = 0;
        int ans = 0;
        while (i < n) {
            if (freq[s.charAt(i)] == 1) {
                while (freq[s.charAt(i)] == 1) {
                    freq[s.charAt(j)]--;
                    j++;
                }
            }
            freq[s.charAt(i)]++;
            ans = Math.max(ans, i - j + 1);
            i++;
        }
        return ans;
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int freq[256] = {};
        int n = s.size();
        int i = 0;
        int j = 0;
        int ans = 0;
        while (i < n) {
            if (freq[s[i]] == 1) {
                while (freq[s[i]] == 1) {
                    freq[s[j]]--;
                    j++;
                }
            }
            freq[s[i]]++;
            ans = max(ans, i - j + 1);
            i++;
        }
        return ans;
    }
};